tests/mcs/resource_manager/server_test.go

```diff
@@ -100,3 +101,39 @@ func TestResourceManagerServer(t *testing.T) {
 		re.Equal("{\"name\":\"pingcap\",\"mode\":1,\"r_u_settings\":{\"ru\":{\"state\":{\"initialized\":false}}}}", string(respString))
 	}
 }
+
+func TestResourceManagerRegister(t *testing.T) {
+	re := require.New(t)
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	cluster, err := tests.NewTestCluster(ctx, 1)
+	defer cluster.Destroy()
+	re.NoError(err)
+
+	err = cluster.RunInitialServers()
+	re.NoError(err)
+
+	leaderName := cluster.WaitLeader()
+	leader := cluster.GetServer(leaderName)
+
+	cfg := rm.NewConfig()
+	cfg.BackendEndpoints = leader.GetAddr()
+	cfg.ListenAddr = "127.0.0.1:8087"
```

Comment:
What does it happen if the address has been bound?

Reply:
I copied from `TestResourceManagerServer`, seems it also has the problem. @lhy1024

Reply:
I will fix it

Reply:
replace it with `tempurl.Alloc()` in #6047